### PR TITLE
Cache Gradle dependency cache to speed up CI

### PR DESCRIPTION
**Pull request by glefloch**
This leverage github action to cache gradle dependencies and downloading everything on each run

### CODE REVIEW

File: .github/workflows/ci-actions-incremental.yml [hunk 1]
```diff
@@ -498,6 +498,13 @@ jobs:
       - name: Extract Maven Repo
         shell: bash
         run: tar -xzf maven-repo.tgz -C ~
+      - name: Cache Gradle Repository
+        id: cache-gradle
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle/caches
+          # refresh cache every month to avoid unlimited growth
+          key: q2gradle-${{ steps.get-date.outputs.date }}
```
Comment (gsmet) [inline .github/workflows/ci-actions-incremental.yml:507]: Just an idea but maybe we could try to use the cache feature integrated in `setup-java@v3`? More details here: https://github.com/actions/setup-java#caching-packages-dependencies

**Review by gsmet — COMMENTED** [review]
Cool idea. I added a small suggestion inline. I wouldn't do that for the existing cache now but we could experiment with this for the newly added one.

**Comment by famod** [comment]
Gradle is also running in the initial build, no? Might be beneficial there as well. But I guess the keys should be different then because I guess the gradle run in the initial build pulls in fewer dependencies than in the test job?

**Comment by famod** [comment]
This is what I mean, from the initial build:
```
2022-06-16T19:51:05.3507692Z [INFO] Quarkus - Maven Plugin ............................. SUCCESS [  7.282 s]
2022-06-16T19:51:05.3508108Z [INFO] Quarkus - Gradle Plugin - Parent ................... SUCCESS [  0.047 s]
2022-06-16T19:51:05.3508595Z [INFO] Quarkus - Gradle Model ............................. SUCCESS [01:22 min]
2022-06-16T19:51:05.3509018Z [INFO] Quarkus - Gradle Plugin ............................ SUCCESS [ 39.042 s]
2022-06-16T19:51:05.3509448Z [INFO] Quarkus - Extension Gradle Plugin .................. SUCCESS [ 37.329 s]
2022-06-16T19:51:05.3509886Z [INFO] Quarkus - Command Line Interface ................... SUCCESS [ 10.434 s]
2022-06-16T19:51:05.3510315Z [INFO] Quarkus - Integration Tests ........................ SUCCESS [  0.053 s]
```

File: .github/workflows/ci-actions-incremental.yml (hunk 1 above)
Comment (glefloch) [reply .github/workflows/ci-actions-incremental.yml]: I updated the branch using the `setup-java@v3`

**Review by gsmet — APPROVED** [review]
Let's try that and see how it goes.

**Comment by gsmet** [comment]
I rebased to drop the merge commit btw.

**Comment by glefloch** [comment]
Thanks, I missed it.

**Comment by glefloch** [comment]
I don't think failure is related, right ?

**Comment by famod** [comment]
I don't think so. That test failed in another PR mine in a similar way.

**Pull request merged**